gdal/frmts/netcdf/netcdfvirtual.cpp

```diff
@@ -0,0 +1,330 @@
+    {
+        nc_enddef(ncid);
+    }
+
+        void netCDFVID::nc_vmap()
+    {
+        nc_set_define_mode();
+
+        for(size_t itr_d = 0; itr_d < dimList.size(); itr_d++)
+        {
+            int realDimID;
+            netCDFVDimension& dim = dimList[itr_d];
+
+            if(!dim.isValid())
+            {
+                continue; // don't do anywork if variable is invalid
+            }
+
+            nc_def_dim(ncid, dim.getName().c_str(), dim.getLen(), &realDimID);
+            dimList[itr_d].setRealID(realDimID);
+        }
+
+        for(size_t itr_v = 0; itr_v < varList.size(); itr_v++)
+        {
+            int realVarID;
+            netCDFVVariable& var = varList[itr_v];
+
+            if(!var.isValid())
+            {
+                continue; // don't do anywork if variable is invalid
+            }
+
+            // Convert each virtual dimID to a physical dimID:
+            std::unique_ptr<int, std::default_delete<int[]>> newdims(new int[var.getDimCount()]);
+            for(int dimct = 0; dimct < var.getDimCount(); dimct++)
+            {
+                newdims.get()[dimct] = virtualDIDToDim(var.getDimIds()[dimct]).getRealID();
+            }
+
+            nc_def_var(ncid, var.getName().c_str(), var.getType(), var.getDimCount(), newdims.get(), &realVarID);
+            var.setRealID(realVarID);
+
+            // Now write each of its attributes
+            for(size_t attrct = 0; attrct < var.getAttributes().size(); attrct++)
+            {
+                var.getAttributes()[attrct]->vsync(ncid, realVarID);
+            }
+
+            var.getAttributes().clear();
+        }
+
+        nc_set_data_mode();
+    }
+
+    /* Enquiry Functions
+     * (For use with enquiry information about virtual entities)
+     */
+    netCDFVVariable& netCDFVID::virtualVIDToVar(int virtualID)
+    {
+        if(virtualID >= static_cast<int>(varList.size()) || virtualID < 0)
+        {
+            throw SG_Exception_NVOOB("virtual variable collection");
+        }
+
+        return varList[virtualID];
+    }
+
+    netCDFVDimension& netCDFVID::virtualDIDToDim(int virtualID)
+    {
+        if(virtualID >= static_cast<int>(dimList.size()) || virtualID < 0)
+        {
+            throw SG_Exception_NVOOB("virtual dimension collection");
+        }
+
+        return dimList[virtualID];
+    }
+
+    int netCDFVID::nameToVirtualVID(std::string& name)
+    {
+        if(nameVarTable.count(name) < 1)
+        {
+            throw SG_Exception_BadMapping(name.c_str(), "variable ID lookup");
+        }
+
+        return nameVarTable.at(name);
+    }
+
+    int netCDFVID::nameToVirtualDID(std::string& name)
+    {
+        if(nameDimTable.count(name) < 1)
+        {
+            throw SG_Exception_BadMapping(name.c_str(), "dimension ID lookup");
+        }
+        return nameDimTable.at(name);
+    }
+
+    /* Attribute writing
+     *
+     */
+
+    void netCDFVID::nc_put_vatt_text(int varid, const char* name, const char* out)
```

Comment:
`out` is a strange name for something const... Shouldn't that be `value` or something similar. Applies to many below similar instances

Reply:
Hahaha indeed a strange name,

this was to make it consistent with the netCDF man page, but yeah we can do value :+1: